src/rust/src/encoder/common.rs

```diff
@@ -0,0 +1,545 @@
+                info!("Unable to write CCD header to file\n");
+                return -1;
+            }
+        }
+
+        ccx_output_format::CCX_OF_SCC => {
+            if write_raw(
+                out.fh,
+                SCC_HEADER.as_ptr() as *const c_void,
+                SCC_HEADER.len() - 1,
+            ) == -1
+            {
+                info!("Unable to write SCC header to file\n");
+                return -1;
+            }
+        }
+
+        ccx_output_format::CCX_OF_SRT | ccx_output_format::CCX_OF_G608 => {
+            if write_bom(ctx, out) < 0 {
+                return -1;
+            }
+        }
+
+        ccx_output_format::CCX_OF_SSA => {
+            if write_bom(ctx, out) < 0 {
+                return -1;
+            }
+
+            if !request_buffer_capacity(ctx, (SSA_HEADER.len() * 3) as c_uint) {
+                return -1;
+            }
+
+            // Create safe slice from buffer pointer and capacity
+            let buffer_slice =
+                unsafe { std::slice::from_raw_parts_mut(ctx.buffer, ctx.capacity as usize) };
+            let text_slice = SSA_HEADER;
+            used = encode_line(ctx, buffer_slice, text_slice.as_ref());
+
+            if used > ctx.capacity {
+                return -1;
+            }
+
+            if write_raw(out.fh, ctx.buffer as *const c_void, used as usize) < used as isize {
+                info!("WARNING: Unable to write complete Buffer\n");
+                return -1;
+            }
+        }
+
+        ccx_output_format::CCX_OF_WEBVTT => {
+            if write_bom(ctx, out) < 0 {
+                return -1;
+            }
+
+            // Calculate total header size
+            for header_line in WEBVTT_HEADER {
+                header_size += header_line.len();
+            }
+
+            if !request_buffer_capacity(ctx, (header_size * 3) as c_uint) {
+                return -1;
+            }
+
+            for header_line in WEBVTT_HEADER {
+                let line_to_write = unsafe {
+                    if ccx_options.enc_cfg.line_terminator_lf == 1 && *header_line == "\r\n" {
+                        "\n"
+                    } else {
+                        header_line
+                    }
+                };
+
+                // Create safe slice from buffer pointer and capacity
+                let buffer_slice =
+                    unsafe { std::slice::from_raw_parts_mut(ctx.buffer, ctx.capacity as usize) };
+                let text_slice = line_to_write.as_bytes();
+                used = encode_line(ctx, buffer_slice, text_slice);
+
+                if used > ctx.capacity {
+                    return -1;
+                }
+
+                if write_raw(out.fh, ctx.buffer as *const c_void, used as usize) < used as isize {
+                    info!("WARNING: Unable to write complete Buffer\n");
+                    return -1;
+                }
+            }
+        }
+
+        ccx_output_format::CCX_OF_SAMI => {
```

Comment:
CCX_OF_SAMI, CCX_OF_SMPTETT, CCX_OF_SIMPLE_XML have similar format just header vaulue is different, you can combine them to single case.